yt_dlp/extractor/dacast.py

```diff
@@ -0,0 +1,122 @@
+import hashlib
+import re
+import time
+import urllib.error
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    float_or_none,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class DacastIE(InfoExtractor):
+    _VALID_URL = r'https?://iframe\.dacast\.com/vod/(?P<user_id>[\w-]+)/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://iframe.dacast.com/vod/acae82153ef4d7a7344ae4eaa86af534/1c6143e3-5a06-371d-8695-19b96ea49090',
+        'info_dict': {
+            'id': '1c6143e3-5a06-371d-8695-19b96ea49090',
+            'ext': 'mp4',
+            'uploader_id': 'acae82153ef4d7a7344ae4eaa86af534',
+            'title': '2_4||Adnexal mass characterisation: O-RADS US and MRI||N. Bharwani, London/UK',
+            'thumbnail': 'https://universe-files.dacast.com/26137208-5858-65c1-5e9a-9d6b6bd2b6c2',
+        },
+        'params': {'skip_download': 'm3u8'},
+    }]
+
+    @staticmethod
+    def _get_url_from_id(content_id):
+        url_type = 'playlist' if '-playlist-' in content_id else 'vod'
+        user_id, media_id = content_id.split(f'-{url_type}-')
+        return f'https://iframe.dacast.com/{url_type}/{user_id}/{media_id}'
+
+    @classmethod
+    def _extract_embed_urls(cls, url, webpage):
+        for content_id in re.findall(
+                r'<script[^>]+\bsrc=["\']https://player\.dacast\.com/js/player\.js\?contentId=([\w-]+-vod-[\w-]+)["\']', webpage):
+            yield DacastIE._get_url_from_id(content_id)
```

Comment:
These could be in a base class:

```suggestion
class DacastBaseIE(InfoExtractor):
    @staticmethod
    def _get_url_from_id(content_id):
        url_type = 'playlist' if '-playlist-' in content_id else 'vod'
        user_id, media_id = content_id.split(f'-{url_type}-')
        return f'https://iframe.dacast.com/{url_type}/{user_id}/{media_id}'

    @classmethod
    def _extract_embed_urls(cls, url, webpage):
        for content_id in re.findall(
                r'<script[^>]+\bsrc=["\']https://player\.dacast\.com/js/player\.js\?contentId=([\w-]+-(?:vod|playlist)-[\w-]+)["\']', webpage):
            yield cls._get_url_from_id(content_id)
```

Not mandatory, just a suggestion

Reply:
in order to move `_extract_embed_urls` to the base class, its regex pattern needs to be limited to 1 url type (`vod` vs `playlist`), or else embeds will be detected for both sub classes and one extractor will raise as unsuitable

Reply:
> in order to move `_extract_embed_urls` to the base class, its regex pattern needs to be limited to 1 url type (`vod` vs `playlist`), or else every embed will be detected by both sub classes and one extractor will raise as unsuitable

Nope. The results of _extract_embed_urls are tested again with _VALID_URL. But your solution works too.

Reply:
I meant that it does this
```sh
[generic] Extracting URL: https://gist.githubusercontent.com/bashonly/7efb606f49f3c6e07ea0327de5a661d1/raw/05a16eac830245ea301fb0a585023bec71e6093c/dacast_playlist_embed.html
[generic] dacast_playlist_embed: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] dacast_playlist_embed: Extracting information
[debug] Looking for embeds
[debug] Identified a Dacast embed
[debug] Identified a DacastPlaylist embed
[download] Downloading playlist: hello
[generic] Playlist hello: Downloading 2 items of 2
[download] Downloading item 1 of 2
ERROR: No suitable extractor (Dacast) found for URL https://iframe.dacast.com/playlist/943bb1ab3c03695ba85330d92d6d226e/b632eb053cac17a9c9a02bcfc827f2d8
  File "/home/user/bin/yt_dlp", line 9, in <module>
    sys.exit(main())
  File "/home/user/git/yt-dlp/yt_dlp/__init__.py", line 969, in main
    _exit(*variadic(_real_main(argv)))
  File "/home/user/git/yt-dlp/yt_dlp/__init__.py", line 959, in _real_main
    return ydl.download(all_urls)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 3369, in download
    self.__download_wrapper(self.extract_info)(
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 3344, in wrapper
    res = func(*args, **kwargs)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1507, in extract_info
    return self.__extract_info(url, self.get_info_extractor(key), download, extra_info, process)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1615, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1744, in process_ie_result
    return self.__process_playlist(ie_result, download)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1888, in __process_playlist
    entry_result = self.__process_iterable_entry(entry, download, collections.ChainMap({
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1920, in __process_iterable_entry
    return self.process_ie_result(
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1694, in process_ie_result
    return self.extract_info(
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1510, in extract_info
    self.report_error(f'No suitable extractor{format_field(ie_key, None, " (%s)")} found for URL {url}',
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 1015, in report_error
    self.trouble(f'{self._format_err("ERROR:", self.Styles.ERROR)} {message}', *args, **kwargs)
  File "/home/user/git/yt-dlp/yt_dlp/YoutubeDL.py", line 944, in trouble
    tb_data = traceback.format_list(traceback.extract_stack())

[download] Downloading item 2 of 2
[DacastPlaylist] Extracting URL: https://iframe.dacast.com/playlist/943bb1ab3c03695ba85330d92d6d226e/b632eb053cac17a9c9a02bcfc827f2d8
[DacastPlaylist] b632eb053cac17a9c9a02bcfc827f2d8: Downloading JSON metadata
[download] Downloading playlist: Archive Sermons
[DacastPlaylist] Playlist Archive Sermons: Downloading 29 items of 29
[download] Downloading item 1 of 29
[Dacast] Extracting URL: https://iframe.dacast.com/vod/943bb1ab3c03695ba85330d92d6d226e/dfb24953-bd3d-42a5-8d8c-a55f91899595
[Dacast] dfb24953-bd3d-42a5-8d8c-a55f91899595: Downloading JSON metadata
[Dacast] dfb24953-bd3d-42a5-8d8c-a55f91899595: Downloading access JSON
^C
ERROR: Interrupted by user
```

it does download the embed, but I'd rather it didn't error first